include/macroni/Dialect/Kernel/KernelOps.td

```diff
@@ -119,14 +119,192 @@ def RCUDereference :
+    @c: The conditions under which the dereference will take place
+    See https://www.kernel.org/doc/html/next/core-api/kernel-api.html
+  }];
+  let assemblyFormat = [{
+    `rcu_dereference_sched_check` `(` $p `,` $c `)` attr-dict
+    `:` functional-type( operands, results )
+  }];
+}
+
+def RCUDereferenceProtected :
+  Kernel_Op< "rcu_dereference_protected", [] >
+  , Arguments<(ins
+    AnyType:$p,
+    AnyType:$c)>
+  , Results<(outs AnyType:$result)> {
+  let summary = [{
+    An expansion of the Linux kernel macro, rcu_dereference_protected()
+  }];
+  let description = [{
+    rcu_dereference_protected() - fetch RCU pointer when updates prevented
+    @p: The pointer to read, prior to dereferencing
+    @c: The conditions under which the dereference will take place
+    See https://www.kernel.org/doc/html/next/core-api/kernel-api.html
+  }];
+  let assemblyFormat = [{
+    `rcu_dereference_protected` `(` $p `,` $c `)` attr-dict
+    `:` functional-type( operands, results )
+  }];
+}
+
+def RCUAccessPointer :
+  Kernel_Op< "rcu_access_pointer", [] >
+  , Arguments<(ins AnyType:$p)>
+  , Results<(outs AnyType:$result)> {
+  let summary = [{
+    An expansion of the Linux kernel macro, rcu_access_pointer()
+  }];
+  let description = [{
+    rcu_access_pointer() - fetch RCU pointer with no dereferencing 
+    @p: The pointer to read
+    See https://www.kernel.org/doc/html/next/core-api/kernel-api.html
+  }];
+  let assemblyFormat = [{
+    `rcu_access_pointer` `(` $p `)` attr-dict
+    `:` functional-type( operands, results )
+  }];
+}
+
+def RCUAssignPointer :
+  Kernel_Op< "rcu_assign_pointer", [] >
+  , Arguments<(ins
+    AnyType:$p
+    , AnyType:$v)>
+  , Results<(outs AnyType:$result)> {
+  let summary = [{
+    An expansion of the Linux kernel macro, rcu_assign_pointer()
+  }];
+  let description = [{
+    rcu_assign_pointer() - assign to RCU-protected pointer
+    @p: The pointer to read
+    @v: The value to statically initialize with.
+    See https://www.kernel.org/doc/html/next/core-api/kernel-api.html
+  }];
+  let assemblyFormat = [{
+    `rcu_assign_pointer` `(` $p `,` $v `)` attr-dict
+    `:` functional-type( operands, results )
+  }];
+}
+
+def RCUReplacePointer :
+  Kernel_Op< "rcu_replace_pointer", [] >
+  , Arguments<(ins
+    AnyType:$rcu_ptr
+    , AnyType:$ptr
+    , AnyType:$c)>
+  , Results<(outs AnyType:$result)> {
+  let summary = [{
+    An expansion of the Linux kernel macro, rcu_replace_pointer()
+  }];
+  let description = [{
+    rcu_replace_pointer() - replace an RCU pointer, returning its old value
+    @rcu_ptr: RCU pointer, whose old value is returned
+    @ptr: regular pointer
+    @c: the lockdep conditions under which the dereference will take place
+    See https://www.kernel.org/doc/html/next/core-api/kernel-api.html
+  }];
+  let assemblyFormat = [{
+    `rcu_replace_pointer` `(` $rcu_ptr `,` $ptr `,` $c `)` attr-dict
+    `:` functional-type( operands, results )
+  }];
+}
+
```

Comment:
There are some repetitions (in arguments/outputs and assembly format) and I think generalisation makes sense here - if unsure how consult how VAST does it.